calibration/test/e2e_test.jl

```diff
@@ -0,0 +1,169 @@
+with top-of-atmosphere radiative shortwave flux in the loss function.
+
+The calibration is run twice, once on the backend obtained via `get_backend()`
+and once on the `JuliaBackend`. The output of each calibration is tested individually
+and compared to ensure reproducibility.
+=#
+import ClimaCalibrate as CAL
+import ClimaAtmos as CA
+import ClimaAnalysis: SimDir, get, slice, average_xy
+import CairoMakie
+import JLD2
+import LinearAlgebra: I
+import EnsembleKalmanProcesses as EKP
+import Statistics: var, mean
+using Test
+
+# Paths and setup
+experiment_dir = joinpath(pkgdir(CA), "calibration", "test")
+model_interface = joinpath(pkgdir(CA), "calibration", "model_interface.jl")
+include(model_interface)
+
+# Observation map
+function CAL.observation_map(iteration)
+    ensemble_size = 10
+    output_dir = "calibration_end_to_end_test"
+    single_member_dims = (1,)
+    G_ensemble = Array{Float64}(undef, single_member_dims..., ensemble_size)
+
+    for m in 1:ensemble_size
+        member_path = CAL.path_to_ensemble_member(output_dir, iteration, m)
+        simdir_path = joinpath(member_path, "output_active")
+        if isdir(simdir_path)
+            simdir = SimDir(simdir_path)
+            G_ensemble[:, m] .= process_member_data(simdir)
+        else
+            G_ensemble[:, m] .= NaN
+        end
+    end
+    return G_ensemble
+end
+
+function process_member_data(simdir::SimDir)
+    isempty(simdir.vars) && return NaN
+    rsut =
+        get(simdir; short_name = "rsut", reduction = "average", period = "30d")
+    return slice(average_xy(rsut); time = 30).data
+end
+
+# Generate observations
+obs_path = joinpath(experiment_dir, "observations.jld2")
+if !isfile(obs_path)
+    @info "Generating observations"
+    config = CA.AtmosConfig(joinpath(experiment_dir, "model_config.yml"))
+    simulation = CA.get_simulation(config)
+    CA.solve_atmos!(simulation)
+    observations = Vector{Float64}(undef, 1)
+    observations .= process_member_data(SimDir(simulation.output_dir))
+    JLD2.save_object(obs_path, observations)
+end
+
+# Initialize experiment data
+astronomical_unit = 149_597_870_000
+observations = JLD2.load_object(obs_path)
+noise = 0.1 * I
+n_iterations = 3
+ensemble_size = 10
+prior = CAL.get_prior(joinpath(experiment_dir, "prior.toml"))
+output_dir = "calibration_end_to_end_test"
+experiment_config = CAL.ExperimentConfig(;
+    n_iterations,
+    ensemble_size,
+    observations,
+    noise,
+    output_dir,
+    prior,
+)
+
+# Minimal EKI test
+function minimal_eki_test(julia_eki)
+    params = EKP.get_u(julia_eki)
+    spread = map(x -> var(abs.(x)), params)
+
+    # Spread should be heavily decreased as particles have converged
+    @test last(spread) / first(spread) < 0.1
+    # Parameter should be close to true value
+    @test mean(last(params)) ≈ astronomical_unit rtol = 0.02
+end
+
+# Caltech HPC backend
+backend = CAL.get_backend()
+@info "Running Calibration E2E test on $backend"
+if backend <: CAL.SlurmBackend
```

Comment:
```suggestion
if backend isa CAL.SlurmBackend
```

Reply:
I believe `<:` is required because the backend is the struct itself (`CaltechHPC`), not an instance of the struct (`CaltechHPCBackend()`).

Reply:
Probably better to use `backend <: CAL.SlurmBackend` if `backend` is a type, to not assume how `backend` is constructed.